sql/core/src/main/scala/org/apache/spark/sql/jdbc/MsSqlServerDialect.scala

```diff
@@ -181,4 +196,20 @@ private object MsSqlServerDialect extends JdbcDialect {
       case _ => super.classifyException(message, e)
     }
   }
+
+  class MsSqlServerSQLQueryBuilder(dialect: JdbcDialect, options: JDBCOptions)
+    extends JdbcSQLQueryBuilder(dialect, options) {
+
+    // TODO[SPARK-42289]: DS V2 pushdown could let JDBC dialect decide to push down offset
```

Comment:
what does this TODO mean?

Reply:
Spark force push down offset to databases, but some databases do not support offset.